pan3d/dataset_builder.py

```diff
@@ -386,12 +411,81 @@ def mesh_updated(exception=None):
+                config = json.load(f)
+        else:
+            config = json.loads(config_file)
+        origin_config = config.get("data_origin")
+        array_config = config.get("data_array")
+        slices_config = config.get("data_slices")
+        ui_config = config.get("ui")
+
+        if not origin_config or not array_config:
+            self.state.dialog_message = "Invalid format of import file."
+            return
+
+        self.set_dataset_path(dataset_path=origin_config)
+        if "active" in array_config:
+            self.set_active_array(array_config["active"])
+        for axis in ["x", "y", "z", "t"]:
+            if axis in array_config:
+                self.state[axis + "_array"] = array_config[axis]
+        if "t_index" in array_config:
+            self.state.t_index = array_config["t_index"]
+
+        if slices_config:
+            coordinates = self.state.coordinates
+            for coordinate in coordinates:
+                if coordinate["name"] in slices_config:
+                    start, stop, step = slices_config[coordinate["name"]]
+                    coordinate["start"] = start
+                    coordinate["stop"] = stop
+                    coordinate["step"] = step
+            self.state.coordinates = coordinates
+            self.state.dirty("coordinates")
+
+        if ui_config:
+            self.state.update(ui_config)
+            self.state.dirty(*ui_config.keys())
 
-    def export_config(self, config_path):
-        # TODO
-        pass
+        self.state.update({"dialog_shown": None, "selected_config_file": None})
+
+    def export_config(self, config_file=None):
+        config = {}
+        config["data_origin"] = self.state.dataset_path
+        config["data_array"] = {"active": self.state.active_array}
+
+        for axis in ["x", "y", "z", "t"]:
+            if self.state[axis + "_array"]:
+                config["data_array"][axis] = self.state[axis + "_array"]
+        if self.state.t_index:
+            config["data_array"]["t_index"] = self.state.t_index
+
+        coordinates = self.state.coordinates
+        for coordinate in coordinates:
+            if (
+                coordinate.get("start")
+                or coordinate.get("stop")
+                or coordinate.get("step")
+            ):
+                if "data_slices" not in config:
+                    config["data_slices"] = {}
+                coordinate_slice = [
+                    coordinate.get("start", 0),
+                    coordinate.get("stop", -1),
+                    coordinate.get("step", 1),
+                ]
+                config["data_slices"][coordinate["name"]] = coordinate_slice
+
+        for state_var in ["main_drawer", "expanded_coordinates"]:
+            if "ui" not in config:
+                config["ui"] = {}
+            config["ui"][state_var] = self.state[state_var]
+
+        if config_file:
+            with open(config_file, "w") as f:
```

Comment:
you can use Path from pathlib to write the file in 1 line.

Reply:
Changed in [ad725e2](https://github.com/Kitware/pan3d/pull/33/commits/ad725e2a63525dc5eb16b2779d48f51ec72d9cae)

Reply:
still not what I was thinking for 
https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text

Reply:
Changed in [b816852](https://github.com/Kitware/pan3d/pull/33/commits/b81685249babc1f58e001ca34c9f7d61bc8db8ea)